TargetComposition/SymBuilder/ApiProvider.cpp

```diff
@@ -1862,6 +1862,22 @@ Object ParametersObject::Add(_In_ const Object& /*parametersObject*/,
     return parameterFactory.CreateInstance(spParameter);
 }
 
+void ParametersObject::PropagateLiveRangesFromCallingConvention(_In_ const Object& /*parametersObject*/,
+                                                                _In_ ComPtr<FunctionSymbol>& spFunctionSymbol)
+{
+    //
+    // Determine the platform default calling convention.  If we do not understand the calling convention,
+    // generate an error.
+    //
+    auto pManager = spFunctionSymbol->InternalGetSymbolSet()->GetSymbolBuilderManager();
+
+    CallingConvention *pConvention;
+    CheckHr(pManager->GetDefaultCallingConvention(&pConvention));
+
+    RangeBuilder builder;
+    builder.PropagateParameterRanges(spFunctionSymbol.Get(), pConvention);
+}
+
 std::experimental::generator<Object> ParametersObject::GetIterator(_In_ const Object& /*parametersObject*/,

```

Comment:
Probably it would be good idea to remove this ref on Object parameter as someone may copy/paste this method to use it as a prototype to another place.

Reply:
That's probably a good thought ;)  I'll make that change and see if that was missed anywhere else in the sample.

Reply:
Removed this from all the generator functions in ApiProvider.cpp.  Changed from "Object (const) &" to "Object" to avoid any reference of unwound memory in the generator function if the parameters were ever referenced.